src/image_ops.rs

```diff
@@ -1210,6 +1212,95 @@ cropping if it does not fit.",
         Example::new(1, 1, "DUP > const 184 268 (255, 255, 0) > overlay 10 50")
 );
 
+//-----------------------------------------------------------------------------
+// Pad
+//-----------------------------------------------------------------------------
+
+#[derive(Debug, Clone, PartialEq)]
+struct Pad {
+    top: u32,
+    right: u32,
+    bottom: u32,
+    left: u32,
+    color: Color,
+}
+
+impl ImageOp for Pad {
+    fn apply(&self, stack: &mut ImageStack) {
+        let over = stack.pop();
+
+        let width = self.left + self.right + over.width();
+        let height = self.top + self.bottom + over.height();
+
+        let under = match self.color {
+            Color::Luma(l) => ImageLuma8(ImageBuffer::from_pixel(width, height, l)),
+            Color::LumaA(l) => ImageLumaA8(ImageBuffer::from_pixel(width, height, l)),
+            Color::Rgb(l) => ImageRgb8(ImageBuffer::from_pixel(width, height, l)),
+            Color::Rgba(l) => ImageRgba8(ImageBuffer::from_pixel(width, height, l)),
+        };
+
+        use DynamicImage::*;
+        let result = match (under, over) {
+            (ImageLuma8(mut under), ImageLuma8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageLuma8(under)
+            }
+            (ImageLumaA8(mut under), ImageLumaA8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageLumaA8(under)
+            }
+            (ImageRgb8(mut under), ImageRgb8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageRgb8(under)
+            }
+            (ImageRgba8(mut under), ImageRgba8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageRgba8(under)
+            }
+            (ImageBgr8(mut under), ImageBgr8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageBgr8(under)
+            }
+            (ImageBgra8(mut under), ImageBgra8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageBgra8(under)
+            }
+            // Due to the call to convert_to_color_space above
+            _ => unreachable!(),
+        };
+
+        stack.push(result);
+    }
+
+    fn signature(&self) -> Option<(usize, usize)> {
+        Some((1, 1))
+    }
+}
+
+impl_parse!(
+    Pad,
+    "pad <top> <right> <bottom> <left> '('COLOR')'",
```

Comment:
I'd have naturally gone with `left top right bottom` here. Is there prior art for this order? (I'm not suggesting you change this, or asking for deep digging to justify this ordering; I was just mildly interested that you picked a different convention to the one I find most natural.)

Reply:
If I'm addressing things in a circular pattern, clockwise seems like the obvious direction, and if you're going clockwise, I intuitively went for starting at 12/0 o'clock. But I'm not married to it if there's a convention some other way.

Reply:
I am informed that CSS uses top right bottom left, which seems pretty robust as far as prior art goes!

Reply:
That seems pretty compelling!